File: 1431. Kids With the Greatest Number of Candies/solution.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
  let result = [];
  let maxCandies = 0;
  for (let index = 0; index < candies.length; index++){
    let candy = candies[index];
    if (candy > maxCandies){
      maxCandies = candy;
    }
  }
  
  for (let index = 0; index < candies.length; index++){
    let candy = candies[index];
    let totalCandy = candy + extraCandies;
    if (totalCandy >= maxCandies){
      result.push(true);
    }
    else{
      result.push(false);
    }
  }
  
  return result;
};

File: 1431. Kids With the Greatest Number of Candies/solution.cpp
class Solution {
public:
  vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
    vector<bool> result;
    int maxCandies = 0;
    for (int index = 0; index < candies.size(); index++){
      int candy = candies[index];
      if (candy > maxCandies){
        maxCandies = candy;
      }
    }
    
    for (int index = 0; index < candies.size(); index++){
      int candy = candies[index];
      int totalCandy = candy + extraCandies;
      if (totalCandy >= maxCandies){
        result.push_back(true);
      }
      else{
        result.push_back(false);
      }
    }
    
    return result;
  }
};